src/group_theory/group_action.lean

```diff
@@ -44,6 +44,24 @@ by rw [smul_smul, u.inv_mul, one_smul]
   (u:α) • (↑u⁻¹:α) • x = x :=
 by rw [smul_smul, u.mul_inv, one_smul]
 
+/-- If a monoid `α` acts on `β`, then each `u : units α` defines a permutation of `β`. -/
+def units.smul_perm_hom : units α →* equiv.perm β :=
+{ to_fun := λ u, ⟨λ x, (u:α) • x, λ x, (↑u⁻¹:α) • x, u.inv_smul_smul, u.smul_inv_smul⟩,
+  map_one' := equiv.ext $ one_smul α,
+  map_mul' := λ u₁ u₂, equiv.ext $ mul_smul (u₁:α) u₂ }
```

Comment:
After pending improvements to `@[simps]`, we'll be able to tell it to use `_apply` instead of `to_fun`, and the first person who'll need these `simp` lemmas will add `@[simps]`. I don't want to add unused code that will become obsolete in a few days.